Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Person/GetNearestDrivers.hs

```diff
@@ -110,3 +121,55 @@ getNearestDrivers cityServiceTiers serviceTiers fromLocLatLong radiusMeters merc
         mkDriverResult mbDefaultServiceTierForDriver person vehicle info dist cityServiceTiersHashMap serviceTier = do
           serviceTierInfo <- HashMap.lookup serviceTier cityServiceTiersHashMap
           Just $ NearestDriversResult (cast person.id) person.deviceToken person.language info.onRide (roundToIntegral dist) vehicle.variant serviceTier (maybe 0 (\d -> d.priority - serviceTierInfo.priority) mbDefaultServiceTierForDriver) serviceTierInfo.isAirConditioned location.lat location.lon info.mode person.clientSdkVersion person.clientBundleVersion person.clientConfigVersion person.clientDevice person.backendConfigVersion person.backendAppVersion
+
+scheduledRideFilter :: (MonadFlow m, MonadTime m, LT.HasLocationService m r, ServiceFlow m r) => DST.CurrentSearchInfo -> Id Merchant -> Id DMOC.MerchantOperatingCity -> Bool -> Bool -> DTC.TransporterConfig -> DriverInfo.DriverInformation -> m Bool
+scheduledRideFilter currentSearchInfo merchantId merchantOpCityId isRental isIntercity transporterConfig driverInfo = do
+  now <- getCurrentTime
+  let minimumScheduledBookingLeadTimeInSecs = intToNominalDiffTime (transporterConfig.minmRentalAndScheduledBookingLeadTimeHours * 3600)
+      scheduledRideFilterExclusionThresholdInSecs = intToNominalDiffTime (transporterConfig.scheduledRideFilterExclusionThresholdHours * 3600)
+      haveScheduled = isJust driverInfo.latestScheduledBooking
+  if
+      | haveScheduled && isIntercity -> return False
+      | haveScheduled && isRental -> return $ canTakeRental driverInfo.latestScheduledBooking now minimumScheduledBookingLeadTimeInSecs
+      | checkUsingTime driverInfo.latestScheduledBooking now scheduledRideFilterExclusionThresholdInSecs -> do
+        dropLoc <- currentSearchInfo.dropLocation & fromMaybeM (InternalError "Error: dropLocation not found in currentSearchInfo.")
+        scheduledPickup <- driverInfo.latestScheduledPickup & fromMaybeM (InternalError "Error: latestScheduledPickup not found in driverInfo.")
+        currentDroptoScheduledPickupDistance <-
+          TMaps.getDistance merchantId merchantOpCityId $
+            TMaps.GetDistanceReq
+              { origin = dropLoc,
+                destination = scheduledPickup,
+                travelMode = Just TMaps.CAR,
+                distanceUnit = Meter
+              }
+        routeDistance <- currentSearchInfo.routeDistance & fromMaybeM (InternalError "Error: routeDistance not found in currentSearchInfo.")
+        avgSpeeds <- transporterConfig.avgSpeedOfVehicle & fromMaybeM (InternalError "Error: avgSpeeds not found in transporterConfig.")
+        let avgSpeedOfSUVInKM = avgSpeeds.suv
```

Comment:
Why SUV?

Reply:
transporterConfig prod data for avg speed is not correct . some vehicles have zero values there . so for safe side i fetched suv and it has a 30km/h speed . i can do it according to vehicles if db values are non zero

Reply:
Yeah, we will fix them in db.